src/wallet/test/coinselector_tests.cpp

```diff
@@ -541,19 +541,20 @@ BOOST_AUTO_TEST_CASE(knapsack_solver_test)
 
           // Again, we only create the wallet once to save time, but we still run the coin selection RUN_TESTS times.
           for (int i = 0; i < RUN_TESTS; i++) {
-            // picking 50 from 100 coins doesn't depend on the shuffle,
+            // picking 50.5 from 100 coins doesn't depend on the shuffle,
             // but does depend on randomness in the stochastic approximation code
-            BOOST_CHECK(testWallet.SelectCoinsMinConf(50 * COIN, filter_standard, vCoins, setCoinsRet , nValueRet, coin_selection_params, bnb_used));
-            BOOST_CHECK(testWallet.SelectCoinsMinConf(50 * COIN, filter_standard, vCoins, setCoinsRet2, nValueRet, coin_selection_params, bnb_used));
+            // Use 50.5 to be sure that BnB doesn't match
+            BOOST_CHECK(testWallet.SelectCoinsMinConf(50.5 * COIN, filter_standard, vCoins, setCoinsRet , nValueRet, coin_selection_params, bnb_used));
+            BOOST_CHECK(testWallet.SelectCoinsMinConf(50.5 * COIN, filter_standard, vCoins, setCoinsRet2, nValueRet, coin_selection_params, bnb_used));
             BOOST_CHECK(!equal_sets(setCoinsRet, setCoinsRet2));
 
             int fails = 0;
             for (int j = 0; j < RANDOM_REPEATS; j++)
             {
-                // selecting 1 from 100 identical coins depends on the shuffle; this test will fail 1% of the time
+                // selecting 0.5 from 100 identical coins depends on the shuffle; this test will fail 1% of the time
```

Comment:
Why does this fail at all? Couldn't we just test that selected the right count of inputs and the right input amount?

Reply:
Because we prefer BnB if it works, the test would previously always use BnB because the coins were always `1 * COIN`. This change (and other similar changes to use a `0.5`) ensures that we cannot exactly match the target with the given coins so the `KnapsackSolver` is always used.

Reply:
I've rewritten this comment.